arch/riscv/core/smp.c

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2021 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <init.h>
+#include <kernel.h>
+#include <ksched.h>
+
+volatile struct {
+	arch_cpustart_t fn;
+	void *arg;
+} riscv_cpu_init[CONFIG_MP_NUM_CPUS];
+
+volatile uint32_t riscv_cpu_wake_flag;
```

Comment:
This should probably be an `uintptr_t` to ensure no problems with the `RV_OP_LOADREG` and `RV_OP_STOREREG` commands in 'arch/riscv/core/reset.S' and non-32-bit platforms.

Reply:
Makes sense, thanks!